src/main/java/com/openshift/internal/restclient/model/DeploymentConfig.java

```diff
@@ -164,4 +190,64 @@ public IDeploymentTrigger addTrigger(String type) {
 	public String getDeploymentStrategyType() {
 		return asString(DEPLOYMENTCONFIG_STRATEGY);
 	}
+
+	@Override
+	public boolean haveTriggersFired() {
+		ModelNode causes = get(DEPLOYMENTCONFIG_CAUSES);
+		if (causes.getType() != ModelType.LIST)
+			return false;
+		return causes.asList().size() > 0;
+	}
+
+	@Override
+	public boolean didImageTrigger(String imageNameTag) {
+		if (!haveTriggersFired() || imageNameTag == null)
+			return false;
+		ModelNode causes = get(DEPLOYMENTCONFIG_CAUSES);
```

Comment:
You may need to compare against ModelType.UNDEFINED for causes where the node doesnt exist